R/matrix_form.R

```diff
@@ -73,15 +85,37 @@ mform_handle_newlines <- function(matform) {
 
     mf_strings(matform) <- newstrmat
     mf_formats(matform) <- newfrmmat
-    mf_spans(matform) <- expand_mat_rows(mf_spans(matform), row_nlines, rep_vec_to_len)
-    mf_aligns(matform) <- expand_mat_rows(mf_aligns(matform), row_nlines, rep_vec_to_len)
+    mf_spans(matform) <- expand_mat_rows(spamat, row_nlines, rep_vec_to_len)
+    mf_aligns(matform) <- expand_mat_rows(alimat, row_nlines, rep_vec_to_len)
 ##    mf_display(matform) <- expand_mat_rows(mf_display(matform), row_nlines, rep_vec_to_len)
-    mf_lgrouping(matform) <- rep(mf_lgrouping(matform), times = row_nlines)
+    mf_lgrouping(matform) <- rep(line_grouping, times = row_nlines)
   }
 
   matform
 }
 
+# Helper function to recompact the lines following line groupings to then have them expanded again
+.compress_mat <- function(mat, line_grouping, collapse_method = c("nl", "unique")) {
+  list_compacted_mat <- lapply(unique(line_grouping), function(lg){
+    apply(mat, 2, function(mat_cols) {
+      col_vec <- mat_cols[which(line_grouping == lg)]
+      if (collapse_method[1] == "nl") {
+        paste0(col_vec, collapse = "\n")
+      } else {
+        val <- unique(col_vec)
+        val <- val[nzchar(val)]
+        if (length(val) > 1) {
+          stop("Problem in linegroupings! Some do not have the same values.") # nocov
+        } else if(length(val) < 1) {
+          val <- "" # Case in which it is only ""
+        }
+        val[[1]]
```

Comment:
These are done for formats (that can be not unique, e.g. `c("1", "")` and `val[[1]]` is needed when there are functions (automatic cast to lists)